packages/reports/addon/components/filter-values/dimension-select.ts

```diff
@@ -110,16 +114,30 @@ export default class DimensionSelectComponent extends Component<DimensionSelectC
    */
   @task({ restartable: true })
   *searchDimensionValues(term: string): TaskGenerator<NaviDimensionModel[] | undefined> {
+    const { dimensionColumn } = this;
     const searchTerm = term.trim();
     this.searchTerm = searchTerm;
     if (!searchTerm) {
       return undefined;
     }
+
     yield timeout(this.isSmallCardinality ? SEARCH_DEBOUNCE_OFFLINE_MS : SEARCH_DEBOUNCE_MS);
     const dimensionResponse: NaviDimensionResponse = yield taskFor(this.naviDimension.search).perform(
       this.dimensionColumn,
       searchTerm
     );
-    return dimensionResponse.values;
+    if (dimensionResponse.values.map((each) => each.value).includes(searchTerm)) {
+      return dimensionResponse.values;
+    }
+
+    const dimensionModelFactory = getOwner(this).factoryFor('model:navi-dimension');
+    const value = searchTerm as string | number;
+    const manualQuery = dimensionModelFactory.create({
```

Comment:
Can we do something like this instead of forcing the value into the existing object? and map the existing dimensionResponse into the same format
```
{
	model: ...,
    meta: {
      manualInputEntry: true
    }
}
```

Reply:
Can we reshape the `manualModel` to be like this?

```js
const manualModel = {
  model: manualQuery,
  meta: {
    manualInputEntry: true,
  },
};
```